apis/v1alpha1/baselineadminnetworkpolicy_types.go

```diff
@@ -0,0 +1,156 @@
+// adminNetworkPolicy API.
+type BaselineAdminNetworkPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Specification of the desired behavior of BaselineAdminNetworkPolicy.
+	Spec BaselineAdminNetworkPolicySpec `json:"spec"`
+
+	// Status is the status to be reported by the implementation.
+	// +optional
+	Status BaselineAdminNetworkPolicyStatus `json:"status,omitempty"`
+}
+
+// BaselineAdminNetworkPolicyStatus defines the observed state of
+// BaselineAdminNetworkPolicy.
+type BaselineAdminNetworkPolicyStatus struct {
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+// BaselineAdminNetworkPolicySpec defines the desired state of
+// BaselineAdminNetworkPolicy.
+type BaselineAdminNetworkPolicySpec struct {
+	// Subject defines the pods to which this BaselineAdminNetworkPolicy applies.
+	Subject AdminNetworkPolicySubject `json:"subject"`
+
+	// List of Ingress rules to be applied to the selected pods if they are not
+	// matched by any AdminNetworkPolicy or NetworkPolicy rules.
+	// A total of 100 Ingress rules will be allowed in each BANP instance.
+	// BANPs with no ingress rules do not affect ingress traffic.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Ingress []BaselineAdminNetworkPolicyIngressRule `json:"ingress,omitempty"`
+
+	// List of Egress rules to be applied to the selected pods if they are not
+	// matched by any AdminNetworkPolicy or NetworkPolicy rules.
+	// A total of 100 Egress rules will be allowed in each BANP instance. BANPs
+	// with no egress rules do not affect egress traffic.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Egress []BaselineAdminNetworkPolicyEgressRule `json:"egress,omitempty"`
+}
+
+// BaselineAdminNetworkPolicyIngressRule describes an action to take on a particular
+// set of traffic destined for pods selected by a BaselineAdminNetworkPolicy's
+// Subject field. The traffic must match both ports and from.
+type BaselineAdminNetworkPolicyIngressRule struct {
+	// Name is an identifier for this rule, that may be no more than 100 characters
+	// in length. This field should be used by the implementation to help
+	// improve observability, readability and error-reporting for any applied
+	// BaselineAdminNetworkPolicies.
+	// +optional
+	// +kubebuilder:validation:MaxLength=100
+	Name string `json:"name,omitempty"`
+
+	// Action specifies the affect this rule will have on matching traffic,
+	// currently the following actions are supported:
+	// Allow: allows the selected traffic
+	// Deny: denies the selected traffic
+	// This field is mandatory.
+	Action BaselineAdminNetworkPolicyRuleAction `json:"action"`
+
+	// Ports allows for matching traffic based on port and protocols.
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
+
+	// List of sources whose traffic this AdminNetworkPolicyRule applies to.
```

Comment:
```suggestion
	// List of sources whose traffic this rule applies to.
```

oh, I missed this in ANP above: "AdminNetworkPolicyRule" is not a type; you'd have to either say "AdminNetworkPolicyIngressRule"/"AdminNetworkPolicyEgressRule", or just say "rule".